x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Vector.java

```diff
@@ -75,4 +75,8 @@ interface Builder extends Releasable {
      * Whether this vector was released
      */
     boolean isReleased();
+
+    byte SERIALIZE_VECTOR_VALUES = 0;
+    byte SERIALIZE_VECTOR_CONSTANT = 1;
+    byte SERIALIZE_VECTOR_ARRAY = 2;
```

Comment:
Worth some comments I think. Worth explaining that we use 0 and 1 for backwards compatibility.